src/ui/components/filters/sortoptionscomponent.js

```diff
@@ -23,31 +23,53 @@ export default class SortOptionsComponent extends Component {
     this.selectedOptionIndex = parseInt(this.core.globalStorage.getState(this.name)) || 0;
     this.options[this.selectedOptionIndex].isSelected = true;
     this.hideExcessOptions = this._config.showMore && this.selectedOptionIndex < this._config.showMoreLimit;
-    this.showReset = this._config.showReset && this.selectedOptionIndex !== 0;
+    this.searchOnChangeIsEnabled = this._config.searchOnChange;
+    this.showResetIsEnabled = this._config.showReset;
+    this.showReset = this.showResetIsEnabled && this.selectedOptionIndex !== 0;
+    this.isNoResults = false;
 
     /**
-     * This component listens to updates to vertical results, and sets its state to it when
-     * an update occurs.
-     * @type {string}
+     * This component should only render if there are search results, so it should listen
+     * to updates to vertical results and handle them accordingly.
      */
     this.core.globalStorage.on('update', StorageKeys.VERTICAL_RESULTS, verticalResults => {
-      if (verticalResults.searchState === SearchStates.SEARCH_COMPLETE) {
-        this.setState(verticalResults);
+      const isSearchComplete = verticalResults.searchState === SearchStates.SEARCH_COMPLETE;
+
+      if (isSearchComplete) {
+        const isNoResults = verticalResults.resultsContext === ResultsContext.NO_RESULTS;
+        this.handleVerticalResultsUpdate(isNoResults);
       }
     });
   }
 
-  setState (data = {}) {
+  handleVerticalResultsUpdate (isNoResults) {
+    const isPreviousSearchResults = !this.isNoResults;
+    this.isNoResults = isNoResults;
+
+    if (isNoResults) {
```

Comment:
it looks like this handler will only be triggered if you do a search that goes to No Results, and not if you go from No Results to a regular search?

Reply:
Just tested this. It hides when there are no results and it re-appears when there are results